### ISSUE TITLE
Node labelling assumes that the node names are the short hostname

### ISSUE THREAD

**Issue by AWSmith0216** [sassoftware/viya4-iac-k8s#53]
In my environment the kubernetes nodes are represented by their FQDN, rather than short hostname:

```
# kubectl get nodes
NAME                        STATUS   ROLES           AGE   VERSION
adsmit-oc1-m1.foo.bar.com   Ready    control-plane   79m   v1.24.3
adsmit-oc1-n1.foo.bar.com   Ready    <none>          78m   v1.24.3
adsmit-oc1-n2.foo.bar.com   Ready    <none>          78m   v1.24.3
adsmit-oc1-n3.foo.bar.com   Ready    <none>          78m   v1.24.3
adsmit-oc1-n4.foo.bar.com   Ready    <none>          78m   v1.24.3
adsmit-oc1-n5.foo.bar.com   Ready    <none>          78m   v1.24.3
```

However, the **roles/kubernetes/node/labels_taints/tasks/labels.yaml** file uses the **ansible_hostname** magic variable, which seems to only resolve to the short hostname. This causes an error similar to:

`failed: [adsmit-oc1-m1.foo.bar.com] (item=launcher.sas.com/prepullImage=sas-programming-environment) => {"ansible_loop_var": "label", "changed": true, "cmd": "kubectl label nodes adsmit-oc1-m1 launcher.sas.com/prepullImage=sas-programming-environment --overwrite \n", "delta": "0:00:00.122628", "end": "2023-03-16 14:11:51.555860", "label": "launcher.sas.com/prepullImage=sas-programming-environment", "msg": "non-zero return code", "rc": 1, "start": "2023-03-16 14:11:51.433232", "stderr": "Error from server (NotFound): nodes \"adsmit-oc1-m1\" not found", "stderr_lines": ["Error from server (NotFound): nodes \"adsmit-oc1-m1\" not found"], "stdout": "", "stdout_lines": []}`

Seems like the task could be improved to first determine if the node names are the short hostname or FQDN, and then use that name during the labelling.

**Comment by AWSmith0216** [comment]
Entered pull request #61 to resolve this.

**Comment by jarpat** [comment]
Closing, released as part of 3.5.0
https://github.com/sassoftware/viya4-iac-k8s/releases/tag/3.5.0
